packages/worker/src/debug-replayer/index.ts

```diff
@@ -0,0 +1,75 @@
+import worker_threads from 'node:worker_threads';
+import { temporal } from '@temporalio/proto';
+import { ReplayWorkerOptions } from '../worker-options';
+import { Worker } from '../worker';
+import { Client } from './client';
+
+async function run(options: ReplayWorkerOptions): Promise<void> {
+  const baseUrl = process.env.TEMPORAL_DEBUGGER_PLUGIN_URL;
+  if (!baseUrl) {
+    throw new Error('Missing TEMPORAL_DEBUGGER_PLUGIN_URL environment variable');
+  }
+  const client = new Client({ baseUrl });
+  const response = await client.get('history');
+  const history = temporal.api.history.v1.History.decode(
+    await Client.readAll(response),
+    Client.contentLength(response)
+  );
+
+  const thread = new worker_threads.Worker(require.resolve('./worker-thread'));
```

Comment:
Not sure about the intended execution model of the debug-replayer, but there might be a risk of leaking threads here...

Reply:
It's meant to be running once per process but I definitely get your concern. I'll make the thread global.